docs(observability): correct TraceQL syntax in query reference and troubleshooting guide (#3274)
giantswarm/docs: src/content/overview/observability/data-management/data-exploration/traceql/index.md

### Analyze service dependencies

Group the spans of a service to see the operations it runs:

```traceql

> {resource.service.name = "api-gateway"} | by(resource.service.name)

Author: **Prose/query mismatch.** The heading says "Group the spans of a service **to see the operations it runs**", but the query groups by `resource.service.name` — the very attribute the `{}` already filters to a single service. That produces one group and reveals nothing about operations. To actually surface the operations, group by the span name intrinsic:

```suggestion
{resource.service.name = "api-gateway"} | by(span:name)
```

Author: Good catch — fixed. Grouping by the already-filtered `resource.service.name` was pointless. I went with your `by(span:name)` framing but, per your terminal-`by()` concern below, added a trailing aggregate so it matches the only documented form: `{resource.service.name = "api-gateway"} | by(span:name) | count() > 1`. Also renamed the heading to "Group a service's operations" so prose and query agree.

- **Start specific**: Begin with service or operation names before adding duration filters
- **Use time ranges**: Always specify time ranges to improve query performance

> - **Limit results**: Set the result limit in Grafana's query editor for exploratory queries. TraceQL has no `limit()` function.

Reviewer: ```suggestion
- **Limit results**: Set a result limit in Grafana's query editor for exploratory queries.
```

### Common patterns

1. **Error investigation**:

   ```traceql
   {span:status = error} | by(resource.service.name) | count()
   ```

2. **Performance analysis**:

   ```traceql

>    {trace:duration > 5s} | by(resource.service.name)

Author: Two things to check on this "Performance analysis" example:

1. **It no longer performs any analysis.** After dropping the (correctly-removed) `| quantile(0.95)`, the query just regroups slow traces by service with no aggregate — it no longer illustrates a performance metric. Consider pairing it with an aggregate that survives, e.g. `| by(resource.service.name) | avg(span:duration)`.
2. **Terminal `by()` — please SME-verify.** Every example in the Tempo spec pairs `by()` with a *following* aggregate (`… | by(resource.service.name) | count() > 1`). A `by()` as the final pipeline stage with nothing after it may not be valid TraceQL. This affects the terminal `by()` here plus lines 198 and 268, and `| by(span:statusMessage)` in the troubleshooting how-to. If terminal `by()` is not accepted by Tempo, these queries will fail to run — worth confirming as part of the accuracy pass.

Author: Both addressed, and I verified the terminal-`by()` question against the Tempo spec — you're right. The [query-construction reference](https://grafana.com/docs/tempo/latest/traceql/construct-traceql-queries/) documents `by()` **only** as `… | by(<attr>) | <aggregate> <comparison>` (its example: `{ status = error } | by(resource.service.name) | count() > 1`), and aggregates likewise only appear with a trailing comparison. No terminal `by()` or bare terminal aggregate is documented as valid. So I made this consistent everywhere rather than just here:

- Performance analysis → `… | by(resource.service.name) | avg(span:duration) > 1s` (real aggregate, as you suggested).
- This example, the error-investigation and span-kind patterns, the capacity-planning `count()`, the "group a service's operations" example, and `| by(span:statusMessage)` in the troubleshooting guide all now carry a trailing aggregate + comparison.

Note this means your inline suggestion on the other thread (`| by(span:name)` on its own) would have hit the same terminal-`by()` issue, so I extended it with `| count() > 1`.
